src/domain/useCase/sslCertificateWatchDog.go

```diff
@@ -0,0 +1,129 @@
+package useCase
+
+import (
+	"log"
+	"time"
+
+	"github.com/speedianet/os/src/domain/dto"
+	"github.com/speedianet/os/src/domain/repository"
+	"github.com/speedianet/os/src/domain/valueObject"
+)
+
+const (
+	SslValidationsPerHour int    = 3
+	OwnershipValidatePath string = "/validateOwnership"
+)
+
+type SslCertificateWatchDog struct {
+	sslQueryRepo   repository.SslQueryRepo
+	sslCmdRepo     repository.SslCmdRepo
+	vhostQueryRepo repository.VirtualHostQueryRepo
+	vhostCmdRepo   repository.VirtualHostCmdRepo
+}
+
+func NewSslCertificateWatchDog(
+	sslQueryRepo repository.SslQueryRepo,
+	sslCmdRepo repository.SslCmdRepo,
+	vhostQueryRepo repository.VirtualHostQueryRepo,
+	vhostCmdRepo repository.VirtualHostCmdRepo,
+) SslCertificateWatchDog {
+	return SslCertificateWatchDog{
+		sslQueryRepo:   sslQueryRepo,
+		sslCmdRepo:     sslCmdRepo,
+		vhostQueryRepo: vhostQueryRepo,
+		vhostCmdRepo:   vhostCmdRepo,
+	}
+}
+
+func (uc SslCertificateWatchDog) createInlineHtmlMapping(
+	vhost valueObject.Fqdn,
+	ownershipHash string,
+) error {
+	path, _ := valueObject.NewMappingPath(OwnershipValidatePath)
+	matchPattern, _ := valueObject.NewMappingMatchPattern("equals")
+	targetType, _ := valueObject.NewMappingTargetType("inline-html")
+	httpResponseCode, _ := valueObject.NewHttpResponseCode(200)
+	inlineHtmlContent, _ := valueObject.NewInlineHtmlContent(ownershipHash)
+
+	inlineHmtlMapping := dto.NewCreateMapping(
+		vhost,
+		path,
+		matchPattern,
+		targetType,
+		nil,
+		nil,
+		&httpResponseCode,
+		&inlineHtmlContent,
+	)
+	return uc.vhostCmdRepo.CreateMapping(inlineHmtlMapping)
+}
+
+func (uc SslCertificateWatchDog) Execute() {
+	vhosts, err := uc.vhostQueryRepo.Get()
+	if err != nil {
+		log.Printf("FailedToGetVhosts: %s", err.Error())
+		return
+	}
+
+	invalidSslVhosts := []valueObject.Fqdn{}
+	for _, vhost := range vhosts {
+		isSslValid := uc.sslQueryRepo.IsSslPairValid(vhost.Hostname)
+		if isSslValid {
+			continue
+		}
+
+		invalidSslVhosts = append(invalidSslVhosts, vhost.Hostname)
+	}
+
+	for _, invalidSslVhost := range invalidSslVhosts {
+		sslPair, err := uc.sslQueryRepo.GetSslPairByHostname(invalidSslVhost)
+		if err != nil {
+			log.Printf("FailedToGetSslPair (%s): %s", invalidSslVhost, err.Error())
+			continue
+		}
```

Comment:
Conforme eu havia falado com o Andreas na call, usar Is na infra tem que ter um motivo muito válido. Você conseguiria fazer isso aqui usando apenas o Get() do sslPair, fazendo o for loop nos pairs e eliminando ou coletando os SSLs cujo CA ou ExpiresAt sejam inválidos/passado. Isso deixaria a sua lógica do "Valid" mais explícita e eliminaria código.

Reply:
Ajustado.